packages/connect-form/src/hooks/use-connect-form.ts

```diff
@@ -411,6 +417,32 @@ export function handleConnectionFormFieldUpdate(
         },
       };
     }
+    case 'remove-proxy-options': {
+      const proxyOptions = [
+        'proxyHost',
+        'proxyPort',
+        'proxyPassword',
+        'proxyUsername',
+      ];
+      proxyOptions.forEach((key) =>
+        updatedSearchParams.delete(key as keyof MongoClientOptions)
```

Comment:
thought: weird that we need to coerce key, was the compiler complaining about `proxyHost` etc not being a `MongoClientOptions`? https://github.com/mongodb/node-mongodb-native/blob/451627a3a5ac93f259410a12b71d87b0911715e1/src/mongo_client.ts#L252

Reply:
I think `proxyOptions` is `string[]`, and `string` is not a valid type – this should be fixable by using `as const` on the `proxyOptions` definition